integration_test.go

```diff
@@ -108,6 +108,44 @@ func TestWrapUnwrap(t *testing.T) {
 	}
 }
 
+func TestWrapV2Unwrap(t *testing.T) {
+	assert := assert.New(t)
+
+	c, err := NewIntegrationTestClient(t)
+	assert.NoError(err)
+
+	ctx := context.Background()
+
+	crk, err := c.CreateKey(ctx, "kptest-crk", nil, false)
+	assert.NoError(err)
+	t.Logf("CRK created successfully: id=%s\n", crk.ID)
+
+	wrapWithKeyVersion, err := c.WrapV2(ctx, crk.ID, nil, nil)
+	assert.NoError(err)
+
+	getKeyVersion, err := c.ListKeyVersions(ctx, crk.ID, nil)
+	assert.NoError(err)
+	assert.EqualValues(getKeyVersion.KeyVersion[0].ID, wrapWithKeyVersion.KeyVersionDetails.ID)
+
+	unwrapped, err := c.Unwrap(ctx, crk.ID, []byte(wrapWithKeyVersion.CipherText), nil)
+	assert.NoError(err)
+	assert.EqualValues(unwrapped, wrapWithKeyVersion.PlainText)
+
+	keys, err = c.GetKeys(context.Background(), 0, 0)
+	assert.NoError(err)
+
+	for _, key := range keys.Keys {
+		if key.Name == "kptest-crk" {
+			_, err := c.DeleteKey(ctx, key.ID, 0)
+			if err != nil {
+				t.Logf("Error deleting key: %s\n", err)
+			} else {
+				t.Logf("Key deleted: id=%s\n", key.ID)
+			}
+		}
+	}
```

Comment:
optional : Since we created only one key in this tc, so there is no need to call GetKeys & then iterate over one key. You can directly delete one key.